File: 品味Java的21種設計模式/dp_cpp/templatemethod/example7/LoginModel.cpp
#include "LoginModel.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace templatemethod
			{
				namespace example7
				{

					std::string LoginModel::getLoginId()
					{
						return loginId;
					}

					void LoginModel::setLoginId(std::string loginId)
					{
						this->loginId = loginId;
					}

					std::string LoginModel::getPwd()
					{
						return pwd;
					}

					void LoginModel::setPwd(std::string pwd)
					{
						this->pwd = pwd;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/templatemethod/example7/LoginModel.java
package cn.javass.dp.templatemethod.example7;
/**
 * 封装进行登录控制所需要的数据
 */
public class LoginModel {
    /**
     * 登录人员的编号，通用的，可能是用户编号，也可能是工作人员编号
     */
    private String loginId;
    /**
     * 登录的密码
     */
    private String pwd;
    public String getLoginId() {
        return loginId;
    }
    public void setLoginId(String loginId) {
        this.loginId = loginId;
    }
    public String getPwd() {
        return pwd;
    }
    public void setPwd(String pwd) {
        this.pwd = pwd;
    }
}
